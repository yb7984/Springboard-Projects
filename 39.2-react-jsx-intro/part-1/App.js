const App = () => <div>
    <FirstComponent />
    <NamedComponent name="name" />
</div>
