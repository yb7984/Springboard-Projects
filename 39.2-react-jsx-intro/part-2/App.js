const App = () => <div>
    <Tweet username="u1" name="U1" date="2021-2-23" message="m1" />
    <Tweet username="u2" name="U2" date="2021-2-23" message="m2" />
    <Tweet username="u3" name="U3" date="2021-2-23" message="m3" />
</div>
