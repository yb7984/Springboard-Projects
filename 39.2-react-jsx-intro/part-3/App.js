const App = () => <div>
    <Person name="U1" age="10" hobbies={[]} />
    <Person name="U22222222" age="18" hobbies={["fishing" , "sail"]} />
    <Person name="U333333" age="23" hobbies={["soccer" , "basketball"]} />
</div>
