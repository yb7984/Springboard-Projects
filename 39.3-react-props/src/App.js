import Pokegame from "./Pokegame";

function App() {
  return <Pokegame />
}

export default App;
