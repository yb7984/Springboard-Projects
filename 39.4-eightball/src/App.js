import EightBall from "./EightBall";

function App() {
  return (
    <div>
      <EightBall />
    </div>
  );
}

export default App;
