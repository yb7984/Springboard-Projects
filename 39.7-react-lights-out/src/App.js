import React from "react";
import Board from "./Board";
import "./App.css";

/** Simple app that just shows the LightsOut game. */

function App() {
  return (
    <div className="App">
      <Board nrows={10} ncols={10} chanceLightStartsOn={0.9}/>
    </div>
  );
}

export default App;
