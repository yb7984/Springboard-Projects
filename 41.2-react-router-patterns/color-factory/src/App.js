
import { useState } from 'react';
import { Redirect, Route, Switch, useHistory } from 'react-router';
import './App.css';
import Color from './Color';
import ColorForm from './ColorForm';
import ColorList from './ColorList';

function App() {
  const history = useHistory();
  const [colors, setColors] = useState([
    {
      name: "Red",
      value: "#FF0000"
    }
  ]);

  const addColor = (name, value) => {
    setColors(oldColors => {
      return [
        ...oldColors,
        {
          name,
          value
        }
      ];
    });

    history.push('/colors');
  }

  return (
    <div className="App">
      <Switch>
        <Route exact path="/colors/new">
          <ColorForm colors={colors} addColor={addColor}></ColorForm>
        </Route>
        <Route exact path="/colors/:colorName">
          <Color colors={colors}></Color>
        </Route>
        <Route exact path="/colors">
          <ColorList colors={colors}></ColorList>
        </Route>
        <Redirect to="/colors" />
      </Switch>
    </div>
  );
}

export default App;
