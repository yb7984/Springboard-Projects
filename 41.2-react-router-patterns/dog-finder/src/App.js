import './App.css';
import { Route, Switch, Redirect, useRouteMatch, useHistory } from 'react-router';
import DogList from './DogList';
import Nav from './Nav';
import DogDetails from "./DogDetails";
import whiskey from './dogs/whiskey.jpg';
import duke from './dogs/duke.jpg';
import perry from './dogs/perry.jpg';
import tubby from './dogs/tubby.jpg';

function App({ dogs }) {
  const history = useHistory();
  let dogHtml = null;

  //get the dog's name if match detail route
  const match = useRouteMatch('/dogs/:dogName');
  if (match) {
    const dog = dogs.find(dog => (dog.name === match.params.dogName));

    if (dog) {
      dogHtml = <DogDetails dog={dog}></DogDetails>;
    } else {
      history.push('/dogs');
    }
  }
  return (
    <div className="App">
      <Nav dogs={dogs} />
      <Switch>
        <Route path='/dogs/:dogName'>
          {dogHtml}
        </Route>
        <Route path='/dogs'>
          <DogList dogs={dogs}></DogList>
        </Route>
        <Redirect to="/dogs" />
      </Switch>
    </div>
  );
}
App.defaultProps = {
  dogs: [
    {
      name: "Whiskey",
      age: 5,
      src: whiskey,
      facts: [
        "Whiskey loves eating popcorn.",
        "Whiskey is a terrible guard dog.",
        "Whiskey wants to cuddle with you!"
      ]
    },
    {
      name: "Duke",
      age: 3,
      src: duke,
      facts: [
        "Duke believes that ball is life.",
        "Duke likes snow.",
        "Duke enjoys pawing other dogs."
      ]
    },
    {
      name: "Perry",
      age: 4,
      src: perry,
      facts: [
        "Perry loves all humans.",
        "Perry demolishes all snacks.",
        "Perry hates the rain."
      ]
    },
    {
      name: "Tubby",
      age: 4,
      src: tubby,
      facts: [
        "Tubby is really stupid.",
        "Tubby does not like walks.",
        "Angelina used to hate Tubby, but claims not to anymore."
      ]
    }
  ]
}

export default App;
