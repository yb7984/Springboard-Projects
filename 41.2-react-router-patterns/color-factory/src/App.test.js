import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import App from './App';


let alertMsg = null;

window.alert = (msg) => {
  alertMsg = msg;
}

it('renders without crashing', () => {
  render(<MemoryRouter><App /></MemoryRouter>);
});

it("matches snapshot", function () {
  const { asFragment } = render(<MemoryRouter><App /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});


it("matches snapshot, /colors", function () {
  const { asFragment, queryByText } = render(
    <MemoryRouter initialEntries={["/colors"]}>
      <App />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();

  expect(queryByText('Red')).toBeInTheDocument();
  expect(queryByText('Add a color')).toBeInTheDocument();
});



it("matches snapshot, /colors/Red", function () {
  const { asFragment, queryByText } = render(
    <MemoryRouter initialEntries={["/colors/Red"]}>
      <App />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(queryByText('This is Red')).toBeInTheDocument();
});


it("matches snapshot, /colors/Blue, redirect to /colors", function () {
  const { asFragment, queryByText, debug } = render(
    <MemoryRouter initialEntries={["/colors/Blue"]}>
      <App />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();

  expect(queryByText('This is Blue')).not.toBeInTheDocument();
  expect(queryByText('Red')).toBeInTheDocument();
  expect(queryByText('Add a color')).toBeInTheDocument();
});

it("matches snapshot, /colors/new", function () {
  const { asFragment, queryByLabelText } = render(
    <MemoryRouter initialEntries={["/colors/new"]}>
      <App />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();

  expect(queryByLabelText('Color Name:')).toBeInTheDocument();
  expect(queryByLabelText('Color Value:')).toBeInTheDocument();
});



it("/colors, works with the link to color", function () {
  const { queryByText } = render(
    <MemoryRouter initialEntries={["/colors"]}>
      <App />
    </MemoryRouter>
  );

  fireEvent.click(queryByText('Red'));

  expect(queryByText('Red')).not.toBeInTheDocument();
  expect(queryByText("This is Red")).toBeInTheDocument();

  fireEvent.click(queryByText('Go back!'));

  //works with go back link
  expect(queryByText("Red")).toBeInTheDocument();
});


it("/colors, works with the link to new color", function () {
  const { queryByText, queryByLabelText } = render(
    <MemoryRouter initialEntries={["/colors"]}>
      <App />
    </MemoryRouter>
  );

  fireEvent.click(queryByText('Add a color'));

  expect(queryByText('Add a color')).not.toBeInTheDocument();
  expect(queryByLabelText("Color Name:")).toBeInTheDocument();

  fireEvent.click(queryByText('Go back!'));

  //works with go back link
  expect(queryByText("Red")).toBeInTheDocument();


  fireEvent.click(queryByText('Add a color'));

  fireEvent.change(queryByLabelText('Color Name:'), { target: { value: "Blue" } });
  fireEvent.change(queryByLabelText('Color Value:'), { target: { value: "#0000ff" } });

  fireEvent.click(queryByText("Add Color!"));

  //Blue is list
  expect(queryByText("Blue")).toBeInTheDocument();


  // not works when the color already list
  fireEvent.click(queryByText('Add a color'));

  fireEvent.change(queryByLabelText('Color Name:'), { target: { value: "Blue" } });
  fireEvent.change(queryByLabelText('Color Value:'), { target: { value: "#0000ff" } });

  expect(alertMsg).toEqual(null);

  fireEvent.click(queryByText("Add Color!"));

  //show alert message and stay with the form
  expect(alertMsg).toEqual("Color Name has been used, please choose another one!");

  expect(queryByLabelText('Color Name:')).toBeInTheDocument();
});