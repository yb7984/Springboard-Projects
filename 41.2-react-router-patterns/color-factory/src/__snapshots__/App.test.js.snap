// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/colors, works with the links 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="ColorList"
    >
      <div
        class="ColorList-header"
      >
        <h1>
          Welcome to the color factory
        </h1>
        <h2>
          <a
            href="/colors/new"
          >
            Add a color
          </a>
        </h2>
      </div>
      <div>
        <h1>
          Please select a color:
        </h1>
        <div
          class="ColorList-colors"
        >
          <a
            href="/colors/Red"
          >
            Red
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="ColorList"
    >
      <div
        class="ColorList-header"
      >
        <h1>
          Welcome to the color factory
        </h1>
        <h2>
          <a
            href="/colors/new"
          >
            Add a color
          </a>
        </h2>
      </div>
      <div>
        <h1>
          Please select a color:
        </h1>
        <div
          class="ColorList-colors"
        >
          <a
            href="/colors/Red"
          >
            Red
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, /colors 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="ColorList"
    >
      <div
        class="ColorList-header"
      >
        <h1>
          Welcome to the color factory
        </h1>
        <h2>
          <a
            href="/colors/new"
          >
            Add a color
          </a>
        </h2>
      </div>
      <div>
        <h1>
          Please select a color:
        </h1>
        <div
          class="ColorList-colors"
        >
          <a
            href="/colors/Red"
          >
            Red
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, /colors/Blue 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="Color"
      style="background-color: rgb(255, 0, 0);"
    >
      <h1>
        This is Red
      </h1>
      <h1>
        Is it beautiful?
      </h1>
      <h1>
        <a
          href="/colors"
        >
          Go back!
        </a>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, /colors/Blue, redirect to /colors 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="ColorList"
    >
      <div
        class="ColorList-header"
      >
        <h1>
          Welcome to the color factory
        </h1>
        <h2>
          <a
            href="/colors/new"
          >
            Add a color
          </a>
        </h2>
      </div>
      <div>
        <h1>
          Please select a color:
        </h1>
        <div
          class="ColorList-colors"
        >
          <a
            href="/colors/Red"
          >
            Red
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, /colors/Red 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="Color"
      style="background-color: rgb(255, 0, 0);"
    >
      <h1>
        This is Red
      </h1>
      <h1>
        Is it beautiful?
      </h1>
      <h1>
        <a
          href="/colors"
        >
          Go back!
        </a>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, /colors/new 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="ColorForm"
    >
      <form>
        <div
          class="ColorForm-field"
        >
          <label
            for="name"
          >
            Color Name:
          </label>
          <input
            id="name"
            name="name"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="ColorForm-field"
        >
          <label
            for="value"
          >
            Color Value:
          </label>
          <input
            id="value"
            name="value"
            required=""
            type="color"
            value="#FFFFFF"
          />
        </div>
        <button>
          Add Color!
        </button>
      </form>
      <h1>
        <a
          href="/colors"
        >
          Go back!
        </a>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
